react/react.d.ts

```diff
@@ -13,6 +13,7 @@ declare namespace __React {
 
     type Key = string | number;
     type Ref<T> = string | ((instance: T) => any);
+    type State = {} | void;
```

Comment:
Not a huge fan of this alias—to users the type `React.State = {} | void` is potentially confusing. I would prefer to be explicit and use `{} | void` everywhere.